src/engine/EventManager.cxx

```diff
@@ -561,7 +567,7 @@ void EventManager::checkEvents(SDL_Event &event, Engine &engine)
         else if (!tileToPlace.empty() && m_placementAllowed)
         {
           // if targetObject.size > 1 it is a tile bigger than 1x1
-          if (targetObjectNodes.size() > 1 && isPointWithinMapBoundaries(targetObjectNodes))
+          if (targetObjectNodes.size() > 1)
           {
             // instead of using "nodesToPlace" which would be the origin-corner coordinate, we need to pass ALL occupied nodes for now.
             engine.setTileIDOfNode(targetObjectNodes.begin(), targetObjectNodes.end(), tileToPlace, false);
```

Comment:
If I understand correctly, you kind of introduced a bug, which does not manifest due to the setTileIDOfNode function.
You are trying to place half of the object in case it is not on the map (Some of the nodes are filtered out, but you continue with placement),

Reply:
no this check now happens a few line above
<img width="1098" alt="image" src="https://user-images.githubusercontent.com/7029836/159424075-cf20af61-9a45-4e3c-95df-667dccf5f956.png">

Reply:
Difference is following. Previous you check all nodes before place building right? So ether all or none.
In you new code you discard nodes which are not in map, but will try to place rest of them (like half of the building) This will occur in case you are on the upper left map edge).

Reply:
oh, now I get it. I've added a check, if one coord is outside of map boundaries, vector is cleared and aborted.
```cpp
   for (auto coord : engine.map->getObjectCoords(mouseIsoCoords, tileToPlace))
      {
        if (coord.isWithinMapBoundaries())
        {
          targetObjectNodes.push_back(coord);
        }
        else // if one coordinate is outside of mapboundaries, the placement is invalid. Clear the vector and abort
        {
          targetObjectNodes.clear();
          break;
        }
      }
```